### PR TITLE
Add get apps api

### PR DESCRIPTION
**Pull request by duanxianbo**
[(task:19908)](https://staging.teamyapp.com/teams/107/tasks/19908)

### CODE REVIEW

**Comment by duanxianbo** [comment]
need to update cloud

File: core/dao/sqldb/sql.go [hunk 1]
```diff
@@ -13,3 +13,12 @@ func toIDsString(ids []uint64) string {
...
 }
+
+func toStrsString(strs []string) string {
+	strStrings := make([]string, 0)
+	for _, singleStr := range strs {
+		strStrings = append(strStrings, singleStr)
+	}
+
+	return strings.Join(strStrings, ",")
+}
```
Comment (naturetime10) [inline core/dao/sqldb/sql.go]:
```suggestion
```

File: core/dao/sqldb/app_tag_relation.go [hunk 2]
```diff
@@ -18,6 +18,42 @@ type AppTagRelation struct {
 
 var _ dao.AppTagRelation = (*AppTagRelation)(nil)
 
+func (*AppTagRelation) FindAppIDsByTagValuesWithTx(
+	ct context.Context,
+	tx *transaction.Transaction,
+	tagValues []string,
+) ([]uint64, *errs.Error) {
+	tagsStr := toStrsString(tagValues)
```
Comment (naturetime10) [inline core/dao/sqldb/app_tag_relation.go]:
```suggestion
	tagsStr := strings.Join(tagValues, ",")
```

File: core/service/app.go [hunk 3]
```diff
@@ -1191,7 +1246,12 @@ func (a App) uploadAppPackageFiles(
 			headerName := a.removeTarFilePrefix(header.Name)
 			err := a.processFile(ct, userID, appID, versionNumber, tarReader, headerName, func(reader io.Reader) *errs.Error {
 				storageMapKey := path.Join(appPackageRoot, appIDStr, versionNumberStr, headerName)
-				return a.objectStore.Put(ct, storageMapKey, reader)
+				ext := filepath.Ext(headerName)
+				mimeType := mime.TypeByExtension(ext)
+
```
Comment (naturetime10) [inline core/service/app.go]:
```suggestion
```

**Review by naturetime10 — APPROVED** [review]

File: core/service/app.go
Comment (naturetime10) [inline core/service/app.go:1]:
```
AppName        string         `yaml:"appName"`
Description    string         `yaml:"description"`
HasUiExtension bool           `yaml:"hasUiExtension"`
Changes        []string       `yaml:"changes"`
Prices         map[string]int `yaml:"prices"`
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: core/service/app.go
Comment (duanxianbo) [reply core/service/app.go:1]: what is this?

Comment (naturetime10) [reply core/service/app.go:1]: Sorry I couldn't add the comment inline. It's inside `processManifestFile`.

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**